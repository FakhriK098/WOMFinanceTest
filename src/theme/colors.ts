export const colors = {
  primary: '#4F46E5',
  danger: '#DC2626',
  white: '#FFFFFF',
  black: '#000000',
  text: {
    light: '#111827',
    dark: '#F9FAFB',
    mutedLight: '#6B7280',
    mutedDark: '#9CA3AF',
  },
  background: {
    light: '#FFFFFF',
    dark: '#0B1220',
  },
  border: {
    light: '#E5E7EB',
    dark: '#374151',
  },
  inputBg: {
    light: '#1118270D',
    dark: '#111827',
  },
  overlay: 'rgba(0,0,0,0.2)',
}
