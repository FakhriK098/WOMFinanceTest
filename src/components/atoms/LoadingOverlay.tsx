import React from 'react';
import { ActivityIndicator, StyleSheet, View } from 'react-native';
import { colors } from '../../theme/colors';

const LoadingOverlay = () => {
  return (
    <View style={styles.wrapper}>
      <ActivityIndicator color="#fff" />
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.overlay,
  },
});

export default React.memo(LoadingOverlay);
