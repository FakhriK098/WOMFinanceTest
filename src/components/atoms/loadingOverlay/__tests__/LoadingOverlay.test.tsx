import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { ActivityIndicator } from 'react-native';

import LoadingOverlay from '../LoadingOverlay';
import { colors } from '@theme/colors';

describe('LoadingOverlay atom', () => {
  it('matches snapshot', () => {
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(<LoadingOverlay />);
    });
    expect(tr!.toJSON()).toMatchSnapshot();
    act(() => tr!.unmount());
  });

  it('renders activity indicator', () => {
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(<LoadingOverlay />);
    });
    const loaders = tr!.root.findAllByType(ActivityIndicator);
    expect(loaders.length).toBe(1);
    act(() => tr!.unmount());
  });

  it('applies overlay background color', () => {
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(<LoadingOverlay />);
    });
    const wrapper = tr!.root.find((inst) => inst.props && inst.props.style);
    const style = wrapper.props.style as any;
    const flattened = Array.isArray(style) ? style : [style];
    const hasOverlay = flattened.some(
      (s) => s && s.backgroundColor === colors.overlay,
    );
    expect(hasOverlay).toBe(true);
    act(() => tr!.unmount());
  });
});
