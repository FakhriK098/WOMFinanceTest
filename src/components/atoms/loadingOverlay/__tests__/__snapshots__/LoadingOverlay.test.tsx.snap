// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingOverlay atom matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.2)",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <ActivityIndicator
    color="#fff"
  />
</View>
`;
